packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -115,84 +100,81 @@ export function parseTrigger(
-    throw new TypeError(
-      'Could not have inferred the notification trigger type: if you want to use a time interval trigger, pass in only `seconds` with or without `repeats` property; if you want to use calendar-based trigger, pass in `second`.'
-    );
-  } else if ('seconds' in userFacingTrigger) {
-    return {
-      type: 'timeInterval',
-      channelId: userFacingTrigger.channelId,
-      seconds: userFacingTrigger.seconds,
-      repeats: userFacingTrigger.repeats ?? false,
-    };
-  } else if (isCalendarTrigger(userFacingTrigger)) {
-    const { repeats, ...calendarTrigger } = userFacingTrigger;
-    return { type: 'calendar', value: calendarTrigger, repeats };
-  } else {
-    return Platform.select({
-      default: null, // There's no notion of channels on platforms other than Android.
-      android: { type: 'channel', channelId: userFacingTrigger.channelId },
-    });
+  const dateTrigger = parseDateTrigger(userFacingTrigger);
+  if (dateTrigger) {
+    return dateTrigger;
   }
+  const calendarTrigger = parseCalendarTrigger(userFacingTrigger);
+  if (calendarTrigger) {
+    return calendarTrigger;
+  }
+  const dailyTrigger = parseDailyTrigger(userFacingTrigger);
+  if (dailyTrigger) {
+    return dailyTrigger;
+  }
+  const weeklyTrigger = parseWeeklyTrigger(userFacingTrigger);
+  if (weeklyTrigger) {
+    return weeklyTrigger;
+  }
+  const yearlyTrigger = parseYearlyTrigger(userFacingTrigger);
+  if (yearlyTrigger) {
+    return yearlyTrigger;
+  }
+  const timeIntervalTrigger = parseTimeIntervalTrigger(userFacingTrigger);
+  if (timeIntervalTrigger) {
+    return timeIntervalTrigger;
+  }
+  return Platform.select({
+    default: null, // There's no notion of channels on platforms other than Android.
+    android: {
+      type: 'channel',
+      channelId:
+        typeof userFacingTrigger === 'object' &&
+        userFacingTrigger !== null &&
+        !(userFacingTrigger instanceof Date)
+          ? userFacingTrigger?.channelId
+          : undefined,
+    },
+  });
 }
 
-function isCalendarTrigger(
-  trigger: CalendarTriggerInput | ChannelAwareTriggerInput
-): trigger is CalendarTriggerInput {
-  const { channelId, ...triggerWithoutChannelId } = trigger;
-  return Object.keys(triggerWithoutChannelId).length > 0;
-}
-
-function isDateTrigger(
-  trigger:
-    | DateTriggerInput
-    | WeeklyTriggerInput
-    | DailyTriggerInput
-    | CalendarTriggerInput
-    | TimeIntervalTriggerInput
-): trigger is DateTriggerInput {
-  return (
-    trigger instanceof Date ||
-    typeof trigger === 'number' ||
-    (typeof trigger === 'object' && 'date' in trigger)
-  );
+function parseCalendarTrigger(
+  trigger: NotificationTriggerInput
+): NativeCalendarTriggerInput | undefined {
+  if (
+    trigger !== null &&
+    typeof trigger === 'object' &&
+    'type' in trigger &&
+    trigger.type === SchedulableTriggerInputTypes.CALENDAR
+  ) {
+    const { repeats, ...calendarTrigger } = trigger;
```

Comment:
all the other triggers may contain `channelId`; this one does not - is that intended?